src/idlcxx/src/backendCpp11Type.c

```diff
@@ -114,9 +114,22 @@ struct_generate_attributes(idl_backend_ctx ctx)
   idl_indent_incr(ctx);
   for (uint32_t i = 0; i < struct_ctx->member_count; ++i)
   {
-    idl_file_out_printf(ctx, "%s %s_;\n",
-        struct_ctx->members[i].type_name,
-        struct_ctx->members[i].name);
+    const idl_node_t *member_type_node = struct_ctx->members[i].member_type_node;
+    const idl_declarator_t *declarator_node = struct_ctx->members[i].declarator_node;
+    char *defValue = get_default_value(ctx, member_type_node);
+
+    idl_file_out_printf(ctx, "%s %s_",
+      struct_ctx->members[i].type_name,
+      struct_ctx->members[i].name);
+    if (defValue)
+    {
+      if (idl_declarator_is_array((const idl_declarator_t *) declarator_node)) {
+        idl_file_out_printf_no_indent(ctx, " = { }");
+      } else {
+        idl_file_out_printf_no_indent(ctx, " = %s", defValue);
```

Comment:
After building this on my own system, I ran into the following: typedefs of primitives do not have default constructed values
This gives a difference of behaviour if you have a default constructor with side-by-side primitives and typedefs of the same primitives containing different data (the primitives being zeroed, and the typedefs containing garbage).

Reply:
@reicheratwork You are right: I will modify the code to also perform proper initialization to typedefs of (arrays of) primitives.